src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs

```diff
@@ -348,8 +348,16 @@ private static bool IsTargeted(string contextId, double from, double to)
                 hash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes(contextId));
             }
 
+
+            //
+            // The first 4 bytes of the hash will be used for percentage calculation
+            // Endianness check ensures the consistency accross different architectures
+            if (!BitConverter.IsLittleEndian)
+            {
+                Array.Reverse(hash, 0, 4);
+            }
+
             //
-            // Use first 4 bytes for percentage calculation
             // Cryptographic hashing algorithms ensure adequate entropy across hash
             uint contextMarker = BitConverter.ToUInt32(hash, 0);
```

Comment:
I kind of want to explicitly subset the hash to the first 4 bytes just to make things clear. But not needed.

Reply:
Do you mean:

``` c#
byte[] relevantBytes = new byte[4];

Array.Copy(hash, 0, relevantBytes, 0, 4);


if (!BitConverter.IsLittleEndian)
{
    Array.Reverse(relevantBytes);
}

uint contextMarker = BitConverter.ToUInt32(relevantBytes, 0);
```

I used to implement it like this in [30ec65c43b35ebe46ef7c7327b327b150e7a0f18](https://github.com/microsoft/FeatureManagement-Dotnet/pull/405/commits/30ec65c43b35ebe46ef7c7327b327b150e7a0f18)

Reply:
Something like that, but I was hoping for something more like:

```csharp
uint contextMarker = BitConverter.ToUInt32(hash.Take(4), 0);
```

Reply:
That's allocation for no reason.

Reply:
Reason is for the next developer who's reading it to understand it- the way it is now makes it look like we reverse the first 4 bytes but somehow use all of the bytes to determine a resulting uint.

If we'd rather not add the extra array, maybe we could adjust the comments to make it more clear to me. What if we reordered/adjusted them like this:

```csharp
//
// Cryptographic hashing algorithms ensure adequate entropy across hash
using (HashAlgorithm hashAlgorithm = SHA256.Create())
{
    hash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes(contextId));
}

//
// Endianness check ensures the consistency of targeting evaluation result across different architectures
if (!BitConverter.IsLittleEndian)
{
    Array.Reverse(hash, 0, 4);
}

//
// Only the first 4 bytes of the hash will converted to a uint
uint contextMarker = BitConverter.ToUInt32(hash, 0);
```

Reply:
Sure comments, but no need to add allocation for it.